src/command/restore/restore.c

```diff
@@ -1368,14 +1392,23 @@ restoreSelectiveExpression(Manifest *manifest)
                 if (strLstExists(systemDbIdList, includeDb))
                     THROW(DbInvalidError, "system databases (template0, postgres, etc.) are included by default");
 
+                // Error if the db id is in the exclude list
+                if(strLstExists(excludeDbIdList, includeDb))
+                    THROW_FMT(DbInvalidError, "database to include '%s' is in the exclude list", strZ(includeDb));
+
                 // Remove from list of DBs to zero
                 strLstRemove(dbList, includeDb);
             }
 
-            // Exclude the system databases from the list
+            // Remove the system databases from list of DBs to zero unless they are excluded explicitly
             strLstSort(systemDbIdList, sortOrderAsc);
+            systemDbIdList = strLstMergeAnti(systemDbIdList, excludeDbIdList);
```

Comment:
According to stringList.h "The input lists must *both* be sorted ascending or the results will be undefined.". I'm not seeing where `excludeDbIdList` is being sorted before using the `strLstMergeAnti()`.

Reply:
fixed in commit a69c81b, thanks!